instrumentation/spring/spring-jms/spring-jms-2.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/jms/v2_0/SpringTemplateTest.java

```diff
@@ -0,0 +1,272 @@
+    config.setBindingsDirectory(tempDir.getPath());
+    config.setJournalDirectory(tempDir.getPath());
+    config.setCreateBindingsDir(false);
+    config.setCreateJournalDir(false);
+    config.setSecurityEnabled(false);
+    config.setPersistenceEnabled(false);
+    List<CoreQueueConfiguration> list = new ArrayList<>();
+    list.add(new CoreQueueConfiguration("someQueue", "someQueue", null, true));
+    config.setQueueConfigurations(list);
+    Set<TransportConfiguration> set = new java.util.HashSet<>();
+    set.add(new TransportConfiguration(InVMAcceptorFactory.class.getName()));
+    config.setAcceptorConfigurations(set);
+
+    server = HornetQServers.newHornetQServer(config);
+    server.start();
+
+    ServerLocator serverLocator =
+        HornetQClient.createServerLocatorWithoutHA(
+            new TransportConfiguration(InVMConnectorFactory.class.getName()));
+    ClientSessionFactory sf = serverLocator.createSessionFactory();
+    ClientSession clientSession = sf.createSession(false, false, false);
+    clientSession.createQueue("jms.queue.SpringTemplateJms2", "jms.queue.SpringTemplateJms2", true);
+    clientSession.close();
+    sf.close();
+    serverLocator.close();
+
+    HornetQConnectionFactory connectionFactory =
+        HornetQJMSClient.createConnectionFactoryWithoutHA(
+            JMSFactoryType.CF, new TransportConfiguration(InVMConnectorFactory.class.getName()));
+
+    connection = connectionFactory.createConnection();
+    connection.start();
+    session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+    session.run();
+
+    template = new JmsTemplate(connectionFactory);
+    template.setReceiveTimeout(TimeUnit.SECONDS.toMillis(10));
+  }
+
+  @AfterAll
+  static void cleanup() throws Exception {
+    session.close();
+    connection.close();
+    server.stop();
+  }
+
+  @Test
+  void sendingMessageToDestinationNameGeneratesSpans() throws JMSException {
+    Queue queue = session.createQueue("SpringTemplateJms2");
+    template.convertAndSend(queue, messageText);
+    TextMessage receivedMessage = (TextMessage) template.receive(queue);
+
+    assertThat(receivedMessage).isNotNull();
+    assertThat(receivedMessage.getText()).isEqualTo(messageText);
+
+    AtomicReference<SpanData> producerSpan = new AtomicReference<>();
+    testing.waitAndAssertTraces(
+        trace -> {
+          trace.hasSpansSatisfyingExactly(
+              span -> assertProducerSpan(span, "SpringTemplateJms2", false));
+          producerSpan.set(trace.getSpan(0));
+        },
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> {
+                  try {
+                    assertConsumerSpan(
+                        span,
+                        producerSpan.get(),
+                        null,
+                        "SpringTemplateJms2",
+                        "receive",
+                        false,
+                        receivedMessage.getJMSMessageID());
```

Comment:
I'd move this outside of the `waitAndAssertTraces` and save it to a variable to get rid of the try-catch (assuming that this is the method that throws the `JMSException`)